test/jdk/java/awt/print/PrinterJob/XparColor.java

```diff
@@ -75,184 +97,144 @@ public static void main(String s[]) {
         }
     }
 
-    public int print(Graphics g, PageFormat pf, int pi)
-        throws PrinterException {
+    private static void pass() {
+        testPassed = true;
+        countDownLatch.countDown();
+    }
+
+    private static void fail(String failureMsg) {
+        testFailureMsg = failureMsg;
+        testPassed = false;
+        countDownLatch.countDown();
+    }
+
+    private static String convertMillisToTimeStr(int millis) {
+        if (millis < 0) {
+            return "00:00:00";
+        }
+        int hours = millis / 3600000;
+        int minutes = (millis - hours * 3600000) / 60000;
+        int seconds = (millis - hours * 3600000 - minutes * 60000) / 1000;
+        return String.format("%02d:%02d:%02d", hours, minutes, seconds);
+    }
+
+    private static void createAndShowTestDialog() {
+        String testInstruction = "This testcase will be launched twice, once for opengl and once for metal.\n" +
```

Comment:
The test is supposed to be run in all platforms, not only in mac. Since the default pipeline is not opengl in non-mac platform and also metal is not supported in non-mac platform, so this test instruction that it will run in opengl and metal is wrong in non-mac platform. 
You probably need to modify to say it will be run in default pipeline for a particular platform and metal and subsequently check if the platform is non-mac and bail out if metal=true system property is set and inform user that metal cannot be run in present non-mac platform

Reply:
Thanks @prsadhuk, will modify the instruction